core-bundle/src/Routing/ResponseContext/ResponseContext.php

```diff
@@ -12,31 +12,79 @@
 
 namespace Contao\CoreBundle\Routing\ResponseContext;
 
-use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
 
-class ResponseContext implements ResponseContextInterface
+final class ResponseContext
 {
+    public const REQUEST_ATTRIBUTE_NAME = '_contao_response_context';
+
+    /**
+     * @var array<string,\Closure>
+     */
+    private $services;
+
     /**
      * @var PartialResponseHeaderBag|null
      */
     private $headerBag;
 
-    public function getHeaderBag(): ResponseHeaderBag
+    public function add(object $service): self
     {
-        if (null === $this->headerBag) {
-            $this->headerBag = new PartialResponseHeaderBag();
+        return $this->addLazy(\get_class($service), static function () use ($service) { return $service; });
```

Comment:
why would you lazy load the ones that are static? Accessing them would just result in the services key being set?

Reply:
Just reusing the logic that's there. No particular reason.

Reply:
Just looks strange to me, because getting the service then first needs to unwrap the Closure again for no reason…

Reply:
The reason is reusability. That was the quickest way to implement it and it doesn't matter so I just did what I did 🤷‍♂️